packages/datadog-plugin-aws-sdk/src/helpers.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const DEFAULT = 'default'
+const Services = {
+  cloudwatchlogs: require('./services/cloudwatchlogs'),
+  dynamodb: require('./services/dynamodb'),
+  kinesis: require('./services/kinesis'),
+  s3: require('./services/s3'),
+  redshift: require('./services/redshift'),
+  sns: require('./services/sns'),
+  sqs: require('./services/sqs'),
+  [DEFAULT]: require('./services/base')
+}
+
+function getService (serviceName) {
+  if (Services[serviceName]) {
+    return new Services[serviceName]()
+  } else {
+    return new Services[DEFAULT]()
+  }
+}
+
+const helpers = {
+  finish (span, err) {
+    if (err) {
+      span.setTag('error', err)
+
+      if (err.requestId) {
+        span.addTags({ 'aws.response.request_id': err.requestId })
+      }
+    }
+
+    span.finish()
+  },
+
+  addResponseTags (span, response, serviceName, config) {
+    if (!span) return
+
+    if (response.request) {
+      this.addServicesTags(span, response, serviceName)
+    }
+
+    config.hooks.request(span, response)
+  },
+
+  addServicesTags (span, response, serviceName) {
+    if (!span) return
+
+    const params = response.request.params
+    const operation = response.request.operation
+    const service = getService(serviceName)
```

Comment:
This creates a new instance on every request.

Reply:
Maybe instead the `require` could be in `getServices` and then the services variable at the top would become:

```js
services = {
  s3: getService('s3')
}
```

Reply:
Ah, yea that's no good. I've updated this so there's now only 1 instance instantiated for each service. I also tried to introduce the pattern you suggested above, tho i believe this now introduces lazy-loading by wrapping the requires in a function (_which should be fine but just wanted to make sure I was thinking about it correctly_)? also, unless i assign the class that's returned by the `require` to an upper case constant , eslint yells at me,  
```
const services = {
  example_service_name: getService('example_service_name'),
}

function getService (serviceName) {
  const Service = require(`./services/${serviceName}`)
  return new Service()
}
```
, i thought this was proper formatting tho?

Reply:
>  tho i believe this now introduces lazy-loading by wrapping the requires in a function

It doesn't because the function is called synchronously.

> also, unless i assign the class that's returned by the require to an upper case constant , eslint yells at me

That's fine since at that point the variable represents a class and not an object, so it should have an uppercase first character.